components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryRecoveryDataStore.java

```diff
@@ -48,6 +49,8 @@
     public void store(UserRecoveryDataDO recoveryDataDO) throws IdentityException {
         Registry registry = null;
         try {
+            String tenantDomain = IdentityTenantUtil.getTenantDomain(recoveryDataDO.getTenantId());
+            IdentityTenantUtil.initializeRegistry(recoveryDataDO.getTenantId(), tenantDomain);
```

Comment:
To change the initializeRegistry method We'll fix this jira https://wso2.org/jira/browse/IDENTITY-5904